docs: add migration guide for 1.0.2 (#351)
vuestorefront/bigcommerce: docs/releases/migration-guide-v1-0-2.md

        }),
      onComplete,
      onError
    );

  return {
    userData,
    updatePersonalData,
    updatePassword,
    user,
    showEditProfileForm,
    closeEditProfileForm
  };
}
```

## `useCategory` and `useCategoryTree` composables

`useCategory` has been splited into `useCategory` and `useCategoryTree` composables. The `useCategory` can be used to get category information. The `useCategoryTree` can be used to get category tree.

### `AppHeader` component

You should use `useCategoryTree` to build category tree.

```diff
# components/AppHeader.vue

import {
  useCart,
-   useCategory
+   useCategoryTree,
  useUser,
  useWishlist
} from '@vue-storefront/bigcommerce';

...

- const { categories: categoryResults, search: categorySearch } = 
-   useCategory('category-tree');
+ const { categoryTreeItems, load: categorySearch } = 
+   useCategoryTree('category-tree');
const navigation = computed(() =>
-   buildCategoryNavigation(categoryResults.value)
+   buildCategoryNavigation(categoryTreeItems.value)
);

...

const categories = buildSearchCategories(
  products.value.data,
-   categoryResults.value
+   categoryTreeItems.value
);
```

### `Category` page

> You should use `useCategoryTree` to create breadcrumbs and `useCategory` to get category meta information.

Reviewer: ```suggestion
On the Category page, you should use the `useCategoryTree` composable to create breadcrumbs and `useCategory` to get category meta information.
```

    category?.id,
    categories.value,
+    categoryTreeItems.value,
    localePath,
    i18n
  );
  return breadcrumbs;
});

onSSR(async () => {
  isProductsLoading.value = true;
-   await categorySearch();
+   await loadCategoryTree();
  const { categorySlug, page, itemsPerPage, sort, direction } =
    th.getFacetsFromURL();
-   const category = getCategoryBySlug(categorySlug, categories.value);
+   const category = getCategoryBySlug(
+     categorySlug,
+     categoryTreeItems.value
+   );
  await searchCategoriesData({ id: category.id });
  
  ...
});

const metaTags = computed(() => ({
-   title: categoryData.value?.[0]?.page_title || categoryData.value?.[0]?.name,
+   title: categoryData.value?.page_title || categoryData.value?.name,
  meta: [
    {
      hid: 'description',
      name: 'description',
-       content: categoryData.value?.[0]?.meta_description || ''
+       content: categoryData.value?.meta_description || ''
    },
    {
      hid: 'keywords',
      name: 'keywords',
-       content: categoryData.value?.[0]?.meta_keywords || ''
+       content: categoryData.value?.meta_keywords || ''
    }
  ]
}));
```

### `Product` page

> You should use `useCategoryTree` to create breadcrumbs.

Reviewer: ```suggestion
On the Product page, you should use the `useCategoryTree` composable to create breadcrumbs.
```

```diff
# pages/Product.vue

- import { useCategory } from '@vue-storefront/bigcommerce';
+ import { useCategoryTree } from '@vue-storefront/bigcommerce';

...

- const { categories, search: categorySearch } = useCategory('category-tree');
+ const { categoryTreeItems, load: categorySearch } =
+   useCategoryTree('category-tree');

...

const breadcrumbs = computed(() => {
  if (
-     !categories.value ||
-     !categories.value?.length ||
+     !categoryTreeItems.value ||
+     !categoryTreeItems.value?.length ||
    !product.value ||
    !product.value.categories
  ) {
    return [];
  }
  const categoryId = product.value.categories[0];
  const breadcrumbs = getBreadcrumbs(
    categoryId,
-     categories.value,
+     categoryTreeItems.value,
    localePath,
    i18n
  );
  breadcrumbs.push({ text: product.value.name, link: '#' });
  return breadcrumbs;
});
```

## Reviews pagination

> Reviews on `Product` page are now paginated. To add it to your project, you should create `InternalPagination` component, `usePaginationData` composable and modify `Product` page to use them.

Reviewer: ```suggestion
Reviews on the `Product` page are now paginated. To add pagination to your project, you should create the `InternalPagination` component, `usePaginationData` composable, and modify the `Product` page to use them.
```

### `InternalPagination` component

> Create `components/InternalPagination.vue` file with `InternalPagination` component.

Reviewer: ```suggestion
Create the `components/InternalPagination.vue` component with the following content:
```

> ```html

Reviewer: Change `html` to `vue`

        return Array.from(Array(this.total), (_, i) => i + 1);
      },
      limitedPageNumbers() {
        if (this.total <= this.visible) {
          return this.listOfPageNumbers;
        }

        if (
          this.currentPage <
          this.visible - Math.floor(this.visible / 2) + 1
        ) {
          return this.listOfPageNumbers.slice(0, this.visible);
        }

        if (
          this.total - this.currentPage <
          this.visible - Math.ceil(this.visible / 2) + 1
        ) {
          return this.listOfPageNumbers.slice(this.total - this.visible);
        }

        return this.listOfPageNumbers.slice(
          this.currentPage - Math.ceil(this.visible / 2),
          this.currentPage + Math.floor(this.visible / 2)
        );
      },
      firstVisiblePageNumber() {
        return this.limitedPageNumbers[0];
      },
      lastVisiblePageNumber() {
        return this.limitedPageNumbers[this.limitedPageNumbers.length - 1];
      }
    },
    methods: {
      go(page) {
        this.navigate(page);
      },
      getLinkTo(page) {
        return page.toString();
      }
    }
  };
</script>
```

### `usePaginationData` composable

> Create `composables/usePaginationData/index.ts` file with `usePaginationData` composable.

Reviewer: ```suggestion
Create the `composables/usePaginationData/index.ts` file with the `usePaginationData` composable:
```

```js
import { AgnosticPagination } from '@vue-storefront/core';
import { useContext } from '@nuxtjs/composition-api';

export const usePaginationData = () => {
  const { $config } = useContext();

  const getPagination = (meta): AgnosticPagination => {
    if (!meta?.pagination) {
      return {
        currentPage: 1,
        totalPages: 1,
        totalItems: 1,
        itemsPerPage: $config.theme?.itemsPerPage?.[0],
        pageOptions: $config.theme?.itemsPerPage
      };
    }

    return {
      currentPage: meta?.pagination?.current_page,
      totalPages: meta?.pagination?.total_pages,
      totalItems: meta?.pagination?.total,
      itemsPerPage: meta?.pagination?.per_page,
      pageOptions: $config.theme?.itemsPerPage
    };
  };

  return {
    getPagination
  };
};

export default usePaginationData;
```

> ### useProductData composable

Reviewer: ```suggestion
### `useProductData` composable
```

> You should remove `getPagination` from `useProductData` composable, `usePaginationData` is now responsible for that functionality.

Reviewer: ```suggestion
You should remove `getPagination` from `useProductData`, as instead use the new `usePaginationData`  composable that covers their functionalities:
```

-       pageOptions: $config.theme?.itemsPerPage
-     };
-   }
- 
-   return {
-     currentPage: meta?.pagination?.current_page,
-     totalPages: meta?.pagination?.total_pages,
-     totalItems: meta?.pagination?.total,
-     itemsPerPage: meta?.pagination?.per_page,
-     pageOptions: $config.theme?.itemsPerPage
-   };
- };
```

### `Category` page

You should use `usePaginationData` composable for catetegories pagination.

```diff
# pages/Category.vue

+ import { usePaginationData } from '../composables/usePaginationData';

...

const productData = useProductData();
+ const paginationData = usePaginationData();

...

const pagination = computed(() =>
-   productData.getPagination(productsResult.value?.meta)
+   paginationData.getPagination(productsResult.value?.meta)
);

...

return {
  isCategoryGridView,
  th,
  products,
  categoryTree,
  categoryData,
  isProductsLoading,
  pagination,
  activeCategory,
  breadcrumbs,
  wishlist,
  addItemToWishlist,
  removeItemFromWishlist,
  isInWishlist,
  wishlistHelpers,
  addItemToCart,
  isInCart,
  productsQuantity,
  productData,
+   paginationData,
  getPurchasableDefaultVariant,
  isMobile
};
```

### `Product` page

> You should use `InternalPagination` component and `usePaginationData` composable to add product reviews pagination.

Reviewer: ```suggestion
On the Product page, you should use the `InternalPagination` component and `usePaginationData` composable for paginating product reviews:
```

# v1.0.2 migration guide

## What's changed?

- `useUserCredentials` composable has been removed due to security reasons. User credentials is being validated during `updateCustomer` API call.

- `useCategory` has been splited into `useCategory` to get category information and `useCategoryTree` to get category tree.

- Pagination has been added to reviews.

## User validation

> ### PasswordResetForm component

Reviewer: ```suggestion
### `PasswordResetForm` component
```
